### PR TITLE
Type annotations

### LINKED ISSUE #4793: add type annotations
#### Summary

Add some type annotations (https://github.com/nasa/openmct/pull/4789)

The behavior of the app shouldn't change (unless I goofed something), so there's nothing to test specifically, just that the app works.

### PR DESCRIPTION
**Pull request by trusktr**
closes #4793 

### Describe your changes:

- mostly type annotations
- some small code improvements to make types work better, removal of unused args, and small cleanups

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [ ] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [ ] Command line build passes?
* [ ] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/plot/configuration/XAxisModel.js
Comment (trusktr) [inline src/plugins/plot/configuration/XAxisModel.js:48]: Move from https://github.com/nasa/openmct/pull/4710#discussion_r782561369

@shefalijoshi said:

> Actually, let's make these class member methods instead.

File: package.json
Comment (trusktr) [inline package.json:11]: This doesn't update the packages we're currently using, this only adds types (hence the `@types/`) the package's we're using.

File: src/plugins/plot/chart/MCTChartSeriesElement.js
Comment (trusktr) [inline src/plugins/plot/chart/MCTChartSeriesElement.js:91]: What's the point of this `makePoint` call? I left it in place just in case there's a side-effect I don't know about (is there?), but `this.removePoint()` is not using it.

Seems like copy/paste left overs from a long time ago. What's the point of making a point just to remove a another point?

File: src/plugins/plot/chart/MCTChartSeriesElement.js
Comment (trusktr) [inline src/plugins/plot/chart/MCTChartSeriesElement.js:111]: removed another unused arg that didn't do anything.

**Review by unlikelyzero — APPROVED** [review]
This follows the conventions that I've seen with my limited exposure to typescript. I need to defer to someone with more familiarity with the src/api to validate the rest of these changes.

**Comment by unlikelyzero** [comment]
@trusktr added the mocha types from our interactive demo of this functionality during testathon

**Comment by trusktr** [comment]
@unlikelyzero Hey, looks like tests fail because Node is too old for our build setup. My guess is that something had an in-range change that introduced syntax that causes a syntax error in older versions of Node. It'd be nice to be testing on Node 16/17. The new Node 18 LTS is coming out in April.

**Comment by unlikelyzero** [comment]
> @unlikelyzero Hey, looks like tests fail because Node is too old for our build setup. My guess is that something had an in-range change that introduced syntax that causes a syntax error in older versions of Node. It'd be nice to be testing on Node 16/17. The new Node 18 LTS is coming out in April.
> 
> I think this would be a fine change, as our version of Node is what we build with, and not something that impacts consumers. If they want to build from node_modules like Vista does, they'll use whatever build tools and Node version they want.

I don't think there's any particular project keeping us on 12. 18 will be what we need for VIPER Flight. It sounds like there are at least 4 PRs stuck on node 16 support. I think we may need to bite the bullet and figure out why this is failing on node16+ https://github.com/nasa/openmct/issues/3966

File: src/plugins/plot/configuration/XAxisModel.js
Comment (trusktr) [reply src/plugins/plot/configuration/XAxisModel.js:48]: @shefalijoshi what did you mean by this?

**Comment by unlikelyzero** [comment]
@trusktr node16 has been merged in as of monday. I've rebased to pick up that change along with our circleci changes

**Comment by jvigliotta** [comment]
Need to follow up with @akhenry before merge as this will remove lower versions of Node support.

**Comment by trusktr** [comment]
> 14 for open source users

I normally think open source users prefer the latest, and that orgs and corporations are the ones stuck behind.

File: src/plugins/plot/chart/MCTChartSeriesElement.js
Comment (trusktr) [reply src/plugins/plot/chart/MCTChartSeriesElement.js:91]: @shefalijoshi can you take a look at this?

**Comment by trusktr** [comment]
The Node 16 build has the same issue:

```
  An error was thrown in afterAll
  Uncaught Error: Module build failed (from ./node_modules/istanbul-instrumenter-loader/dist/cjs.js):
  SyntaxError: Unexpected token (146:66)
      at Parser.pp$5.raise (/root/project/node_modules/babylon/lib/index.js:4454:13)
      at Parser.pp.unexpected (/root/project/node_modules/babylon/lib/index.js:1761:8)
```

I wonder where that problem is. Wish it would point to the source location, rather than the parser location.

**Comment by trusktr** [comment]
The app works fine though, and the current log-plots stuff is branched off of this, and that works too.

**Comment by jvigliotta** [comment]
@trusktr can you resolve the conflicts? @akhenry can you take a look at this and let us know if we're ok to lose Node 12 support?

**Comment by unlikelyzero** [comment]
@trusktr @akhenry PR To Remove Node 12 support is here: https://github.com/nasa/openmct/pull/4963

File: src/plugins/plot/MctTicks.vue
Comment (trusktr) [inline src/plugins/plot/MctTicks.vue:214]: Our coverage tool has recently been updated, we can freely use modern syntax now.

File: src/plugins/plot/configuration/PlotConfigurationModel.js
Comment (trusktr) [inline src/plugins/plot/configuration/PlotConfigurationModel.js:147]: modern syntax ftw!

**Review by shefalijoshi — APPROVED** [review]
Looks good.

**Pull request merged**
